src/main/java/org/jabref/gui/mergeentries/newmergedialog/fieldsmerger/FileMerger.java

```diff
@@ -25,10 +23,9 @@ public String merge(String filesA, String filesB) {
         } else {
             List<LinkedFile> linkedFilesA = FileFieldParser.parse(filesA);
             List<LinkedFile> linkedFilesB = FileFieldParser.parse(filesB);
-            // TODO: If one of the linked files list is empty then the its string value is malformed.
-            return Stream.concat(linkedFilesA.stream(), linkedFilesB.stream())
```

Comment:
Any explanation why `Stream#concat` didn't work here?

Reply:
I tried it first as well, it seemed a nice solution; however, map will be called on every element in the stream, but we need one method to be called on the whole collection of linkedFiles 
Basically, the method which is called under the hood does the merging:

https://github.com/JabRef/jabref/blob/a022324a85da8e42f7625bfba96ad90796a7cf7e/src/main/java/org/jabref/logic/bibtex/FileFieldWriter.java#L29-L40